62.CAD/main.cpp

```diff
@@ -234,52 +407,866 @@ class CPolyline
 	std::vector<QuadraticBezierInfo> m_quadBeziers;
 };
 
+namespace hatchutils {
+	template<class _Ty, size_t _Size>
+	struct EquationSolveResult
+	{
+		size_t uniqueRoots = 0u;
+		std::array<_Ty, _Size> roots = {};
+	};
+
+	// From https://github.com/erich666/GraphicsGems/blob/master/gems/Roots3And4.c
+	// TODO: Refactor this code (needs better names among other things)
+	// These are not fully safe on NaNs or when handling precision, as pointed out by devsh in the Discord
+	static EquationSolveResult<double, 2> SolveQuadric(double c[3])
```

Comment:
that's Quartic, not Quadric 😄 the original page also has this typo

Reply:
I think that's supposed to be a shortened version